File: examples/python-app/src/python_app/processor.py
from dataclasses import dataclass


@dataclass(frozen=True)
class Report:
	projectName: str
	lineCount: int
	filteredLines: list[str]


def buildReport(projectName: str, lines: list[str], minLineLength: int) -> Report:
	filteredLines = [line for line in lines if len(line) >= minLineLength]
	return Report(
		projectName=projectName,
		lineCount=len(lines),
		filteredLines=filteredLines
	)

File: examples/go-app/internal/processor/processor.go
package processor

type Report struct {
	ProjectName   string
	LineCount     int
	FilteredLines []string
}

func BuildReport(projectName string, lines []string, minLineLength int) Report {
	filteredLines := make([]string, 0, len(lines))
	for _, line := range lines {
		if len(line) >= minLineLength {
			filteredLines = append(filteredLines, line)
		}
	}

	return Report{
		ProjectName:   projectName,
		LineCount:     len(lines),
		FilteredLines: filteredLines,
	}
}
